tasks/configure.yml

```diff
@@ -1,9 +1,20 @@
 ---
-- name: Configure global settings.
+# `data_directory` in postgresql.conf set to `postgresql_data_dir` variable
+# Can be overridden in next task if `data_directory` parameter is also set
+# via `postgresql_global_config_options` var
+- name: Configure data dir in postgresql.conf.
   lineinfile:
     dest: "{{ postgresql_config_path }}/postgresql.conf"
-    regexp: "^#?{{ item.option }}.+$"
-    line: "{{ item.option }} = '{{ item.value }}'"
+    regexp: "^#?\\s*data_directory\\s*=.+$"
+    line: "data_directory = '{{ postgresql_data_dir }}'"
+    mode: 0644
+  notify: restart postgresql
+
+- name: Configure global settings (postgresql.conf).
+  lineinfile:
+    dest: "{{ postgresql_config_path }}/postgresql.conf"
+    regexp: "^#?\\s*{{ item.option }}\\s*=.+$"
+    line: "{{ item.option }} = {{ item.value }}"
```

Comment:
Hi @jeanmonet 

I think this line should be
```
line: "{{ item.option }} = '{{ item.value }}'"
```

Reply:
Well the line you are replacing had it quoted by default.

I wouldn't have though to double quote it like, but that could have just been me.
```yaml
- option: ssl_min_protocol_version
  value: "'TLSv1.3'"
```

Reply:
Just to make it clear. As it is currently proposed you can configure:

```yaml
value: "'TLSv1.3'"   =>   will appear in postgres.conf as    'TSLv1.3'
value: 128MB        =>   will appear in postgres.conf as    128MB
```

Setting `line: "{{ item.option }} = '{{ item.value }}'"` would result in:

```yaml
value: "'TLSv1.3'"   =>   will appear in postgres.conf as    ''TSLv1.3''
value: 128MB        =>   will appear in postgres.conf as    '128MB'
```

Reply:
I guess I was more stating it was a change from how they are defined currently, and maybe would be better to be its own PR since it could be a breaking change?